src/components/carousel/index.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { useState, useEffect, useMemo } from 'react';
+import {
+  CarouselWrapper,
+  CarouselImg,
+  LeftBtn,
+  RightBtn,
+  ProgressBarWrapper,
+  ProgressBar,
+} from './index.style';
+
+interface CarouselProps {
+  images: string[];
+  width?: number;
+  height?: number;
+  showButtons?: boolean;
+  autoPlay?: boolean;
+  autoPlayInterval?: number;
+}
+
+const Carousel: React.FC<CarouselProps> = ({
+  images,
+  width = 480,
+  height = 408,
+  showButtons = true,
+  autoPlay = false,
+  autoPlayInterval = 3000,
+}) => {
```

Comment:
캐러셀 컴포넌트 아주 잘 만들어 주셨네요 👍🏻👍🏻
다만 전에 진행했던 컨벤션 회의에서 컴포넌트는 화살표 함수가 아닌 `function` 키워드를 사용하여 만들기로 했어서, 이 부분 변경해 주시면 좋을 것 같습니다!

```suggestion
export default function Carousel ({
  images,
  width = 480,
  height = 408,
  showButtons = true,
  autoPlay = false,
  autoPlayInterval = 3000,
}) {
```